src/pretix/presale/views/widget.py

```diff
@@ -537,10 +540,21 @@ def _get_event_list(self, request, **kwargs):
             for d in data['days']:
                 d['events'] = self._serialize_events(d['events'] or [])
         else:
+            offset = int(self.request.GET.get("offset", 0))
+            limit = 50
             if hasattr(self.request, 'event'):
                 evs = self.request.event.subevents_sorted(
                     filter_qs_by_attr(self.request.event.subevents_annotated(self.request.sales_channel.identifier), self.request)
                 )
+                ordering = self.request.event.settings.get('frontpage_subevent_ordering', default='date_ascending', as_type=str)
+                data['has_more_events'] = False
+                if ordering in ("date_ascending", "date_descending"):
+                    # fetch one more result than needed to check if more events exist
+                    evs = evs[offset:offset + limit + 1]
+                    if len(evs) > limit:
+                        data['has_more_events'] = True
+                        evs = evs[:limit]
```

Comment:
This will cause more database queries than necessary, because both ``[offset:offset + limit + 1]`` as well as ``[:limit]`` will be executed in SQL.

```suggestion
                    evs = list(evs[offset:offset + limit + 1])
                    if len(evs) > limit:
                        data['has_more_events'] = True
                        evs = evs[:limit]
```
should force early evaluation and fix that

Reply:
Ah, I thought the `len(evs)` would execute the evs-queryset once and after that it’s a list. But your suggestion is more explicit – and so better!